speculos.py

```diff
@@ -63,18 +97,20 @@ def run_qemu(s1, s2, app_path, libraries=[], seed=DEFAULT_SEED, debug=False, tra
     if trace_syscalls:
         args += [ '-t' ]
 
-    if rampage is not None and pagesize is not None:
-        args += [ '-r' , rampage ] + [ '-s' , pagesize ]
-
     if model is not None:
         args += ['-m', model]
 
     args += [ '-k', str(sdk_version) ]
 
     for lib in [ f'main:{app_path}' ] + libraries:
         name, lib_path = lib.split(':')
-        load_offset, load_size, stack, stack_size = get_elf_infos(lib_path)
-        args.append(f'{name}:{lib_path}:{load_offset}:{load_size}:{stack}:{stack_size}')
+        load_offset, load_size, stack, stack_size, ram_addr, ram_size = get_elf_infos(lib_path)
+        # Since binaries loaded as libs could also declare extra RAM page(s), collect them all
+        if (ram_addr, ram_size) != (0, 0):
```

Comment:
If `extra_ram` is a list of tuples, the rest of the code will be much simpler.

Reply:
unless I'm missing something here, it would actually simplify only the following line in `sanitize_ram_args()` function:
```                extra_ram = [dict(s) for s in set(tuple(d.items()) for d in extra_ram)]```
would be changed to:
```                extra_ram = [tuple(s) for s in set(d for d in extra_ram)]```
or something close to that. But in the rest of the code we'd have to address data in extra_ram through indexes. I don't think this change is worth the loss of readability.